cmake/extensions.cmake

```diff
@@ -19,3 +19,17 @@ function(get_board_without_ns_suffix board_in board_out)
     set(${board_out} ${board_in} PARENT_SCOPE)
   endif()
 endfunction()
+
+# Add a kconfig overlay file to a child image.
+# This can be used by a parent image to set kconfig values in its child images.
+# This must be invoked before 'add_child_image(image)'
+function(add_overlay_config image overlay_file)
+  set(old_conf ${${image}OVERLAY_CONFIG})
+  string(FIND "${old_conf}" "${overlay_file}" found)
+  if (${found} EQUAL -1)
+    set(${image}OVERLAY_CONFIG
+      "${old_conf} ${overlay_file}"
```

Comment:
just wondering.
Why is this a space separated string ?
If it was a list, we could simply append to it, and remove duplicates before passing on to child image.
That way we wouldn't need a function for all this, and could go from 18 loc to 2 loc.

Reply:
Hmmm, I'm very sure there was a reason for this, but I don't remember exactly why. Perhaps its because it can be passed as a list from the command line?

Reply:
Isn't it the same for all the other overlays?

Reply:
could be. i think it's worth looking into at later point.
My biggest concern, is that we probably would need a `list(REMOVE_DUPLICATES ...)` later if we change, and I also dislike having specific knowledge of `${image}_VARIABLES` inside `multi_image.cmake` which could be a result of such change.

Reply:
Isn't this the case here as well? Only that we separate on space instead of question mark. Seen in `kconfig.cmake` 

```
# DTS_ROOT_BINDINGS is a semicolon separated list, this causes
# problems when invoking kconfig_target since semicolon is a special
# character in the C shell, so we make it into a question-mark
# separated list instead.
```

Reply:
semicolon needs to be handled with care, but it is possible to pass lists into cmd line using either:
```
--argument "val1;val2;val3"
--argument val1\;val2\;val3
```
Going through some other way is usually a bad sign.
For the sharing of list with child images, that is supposed to be working:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1662
if it does'nt work, blame the reviewer who approved :wink:

Reply:
Discussed offline. List handling like this should be re-assessed upstream.

Reply:
To be looked into as separate cleanup at later time.